### PR TITLE
PS color output fix

### PR DESCRIPTION
**Pull request by Vanello1908**
Fixed one of bugs mentioned in [this](https://github.com/alda-lang/alda/issues/405#issue-1033159114) issue

### CODE REVIEW

**Comment by daveyarwood** [comment]
Hi, thanks for the PR! I tried this branch locally and unfortunately, it doesn't compile on Linux. I get this error:

```
$ go build .
package alda.io/client
        imports alda.io/client/cmd
        imports alda.io/client/color
        imports golang.org/x/sys/windows/registry: build constraints exclude all Go files in /home/dave/.go/pkg/mod/golang.org/x/sys@v0.22.0/windows/registry
```

I did some googling and found [this StackOverflow question](https://stackoverflow.com/questions/72849222/golang-1-18-exclude-package-for-specific-platform) which looks like the same issue.

If we wanted to get this approach to work, I think we would need to use [build constraints](https://pkg.go.dev/go/build#hdr-Build_Constraints) to exclude the Windows registry code when the code is being compiled and run in a non-Windows environment.

I did a little test of this by copy-pasting the code you added into a new file called `color_windows.go`, and it did build successfully after that. However, it won't exactly work that way, because there's a potential race condition where we could end up running `Aurora = auroraLib.NewAurora(EnableColor)` in `color.go` before the code in `color_windows.go` sets `EnableColor` appropriately.

If you'd like to keep working on this approach and adjust it along those lines so that it will compile on Linux, you're welcome to do so. You can check the CircleCI logs to see if the build succeeded, i.e. click the :x: or :heavy_check_mark: next to the commit message:

2024-07-27-173632_1156x324_scrot

And then click the "Details" link to go to the build logs in CircleCI:

2024-07-27-173704_1454x852_scrot

**Comment by daveyarwood** [comment]
My other thought on this is that it would be better if we didn't have to include a bunch of ad hoc code in Alda to work around OS-specific issues. We should really let the color library handle this for us.

The version of the color library we're using (Aurora) is from 2020, which is several years old at this point. I think it's worth trying to update to the latest version and see if it works better out of the box in Powershell.

**Comment by Vanello1908** [comment]
Ok, this is done with build constraint. Next step - black symbols, it is a problem with readline lib. There are 3 options:
- create pull request to that lib with fix
- create fork with fix
- use another lib for input
But I want not to use the third option

File: client/color/color_widows.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+//go:build windows
```
Comment (daveyarwood) [inline client/color/color_widows.go:1]: I think you can remove this line. The Go compiler recognizes when the file name ends with `_windows.go` and only includes it in the build if the OS is Windows.

(You also have a typo in the filename -- should be `color_windows.go`)

File: client/color/color.go [hunk 2]
```diff
@@ -39,7 +41,5 @@ var Aurora auroraLib.Aurora
 func init() {
 	// HACK: Ideally, aurora would support NO_COLOR, but at least they give us a
 	// config option so that we can disable color manually.
-	//
-	// See the longer comment above EnableColor.
```
Comment (daveyarwood) [inline client/color/color.go:43]: Let's put this comment back, as it's still relevant.

File: client/color/color_widows.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	len(os.Getenv("NO_COLOR")) == 0
+
+var Aurora auroraLib.Aurora
+
+func init() {
+	// HACK: Ideally, aurora would support NO_COLOR, but at least they give us a
+	// config option so that we can disable color manually.
+
+	//Check registry for enabled color printing(only for windows)
+	if runtime.GOOS == "windows" {
```
Comment (daveyarwood) [inline client/color/color_widows.go:48]: We probably don't need this check, as this file will only be included in the Windows build.

**Comment by daveyarwood** [comment]
Looking good so far! If you don't mind implementing my suggestions above, I think this will be ready to merge.

I am curious if you've tried updating the Aurora dependency, to see if maybe the library has improved in the last 4 years to the point that none of this OS-specific code is necessary. But I'm also fine with the approach in this PR, at least as an interim improvement.

**Comment by Vanello1908** [comment]
image
aurora v2.0.3+incompatible also has this problems, so some cleanup and last commit :)

**Comment by Vanello1908** [comment]
Aurora v2.0.3+incompatible also has problems with PowerShell, so we remain this code

**Review by daveyarwood — APPROVED** [review]
Looks good! I'm going to go ahead and merge this and cut a patch release of Alda. I appreciate your work on this, and will be happy to review any future PRs to address the remaining Windows color issues.

**Pull request merged**